File: game/scripts/vscripts/items/item_mana_orb/item_mana_orb.lua
item_mana_orb = class({})

function item_mana_orb:OnSpellStart()
    local mana = 8
    local caster = self:GetCaster()
    CustomEntitiesLegacy:GiveManaAndEnergy(caster, mana, true, true)
    
    EmitSoundOn("DOTA_Item.ClarityPotion.Activate", caster)
    self:SpendCharge()
end

File: src/vscripts/items/item_mana_orb.ts
import { CustomItem } from "../abilities/framework/custom_ability";
import { registerAbility } from "../lib/dota_ts_adapter";
import { giveManaAndEnergy } from "../util";

@registerAbility("item_mana_orb")
class ItemManaOrb extends CustomItem {
    OnSpellStart() {
        const mana = 8;
        giveManaAndEnergy(this.GetCaster(), mana, true, true);

        EmitSoundOn("DOTA_Item.ClarityPotion.Activate", this.GetCaster());
        this.SpendCharge();
    }
}
